### PR TITLE
Don't parse SQL if no tracking is in progress

### PR DESCRIPTION
**Pull request by kamil-tekiela**
Surprisingly, performing an SQL query is better for performance than parsing the query unnecessarily. Tested with 250 tables for about 50% better performance. 

Ref. #18249

### CODE REVIEW

**Review by williamdes — APPROVED** [review]
Awesome !

**Comment by liviuconcioiu** [comment]
@kamil-tekiela Thank you for the great work! BTW, this PR will also fix the issues with import of large tables/databases.

In my case, opening a database with 250 tables (50% performance increase) and importing a table (25 MB in size and 25000 rows), before this PR it took 4 minutes and 30 seconds. Now it takes 30 seconds!

File: libraries/classes/Tracker.php [hunk 1]
```diff
@@ -954,4 +958,19 @@ public static function handleQuery($query): void
...
+        string $dbname
+    ): bool {
+        $sqlQuery = sprintf(
+            '/*NOTRACK*/ SELECT 1 FROM %s.%s WHERE tracking_active = 1 AND db_name = %s LIMIT 1',
+            Util::backquote($trackingFeature->database),
+            Util::backquote($trackingFeature->tracking),
+            "'" . $dbi->escapeString($dbname, DatabaseInterface::CONNECT_CONTROL) . "'"
+        );
+
+        return $dbi->queryAsControlUser($sqlQuery)->fetchValue() !== false;
```
Comment (williamdes) [inline libraries/classes/Tracker.php:974]: Is this kept in cache or is it intentional that it is not ?

Comment (kamil-tekiela) [reply libraries/classes/Tracker.php:974]: It's intentional. I thought about caching it but I was unclear if it's a good idea so I intentiontionally left it as is. If someones wants to evaluate the possibility of caching this in the future, they can do it in another PR.

**Pull request merged**